src/System.IO.FileSystem.Watcher.Polling/System/IO/FileSystemChangesEnumerator.cs

```diff
@@ -32,7 +32,12 @@ protected override bool ShouldIncludeEntry(ref FileSystemEntry entry)
             if (_currentDirectory == null)
                 _currentDirectory = entry.Directory.ToString();
 
-            return _watcher.IsWatched(ref entry);
+            return _watcher.ShouldIncludeEntry(ref entry);
+        }
+
+        protected override bool ShouldRecurseIntoEntry(ref FileSystemEntry entry)
+        {
+            return _watcher.ShouldRecurseIntoEntry(ref entry);
```

Comment:
nit: I think it would be better to use an expression bodied method here. It's an one liner after all.